### PR TITLE
[LETS-107] prevent some tools to be executed when 'remote_storage' param is set

### PR DESCRIPTION
**Pull request by cristiarg**
http://jira.cubrid.org/browse/LETS-107

`copydb`, `backupdb`, `restoredb` and `emergency_patchlog` tools run on the disk database files. The tool is not applicable to a transaction server running on remote storage. An error is returned.

**Implementation**
- added new error code & message
- prevented tools to be executed by patching corresponding `xboot_*` functions

**Tests**
- manual tests

### CODE REVIEW

File: src/base/server_type.cpp [hunk 1]
```diff
@@ -83,8 +83,18 @@ void finalize_server_type ()
 
 int init_server_type (const char *)
 {
-  g_server_type = SERVER_TYPE_TRANSACTION;
-  return NO_ERROR;
+  int err_code = NO_ERROR;
+
+  g_server_type = (SERVER_TYPE) prm_get_integer_value (PRM_ID_SERVER_TYPE);
```
Comment (ribram) [inline src/base/server_type.cpp]: SA_MODE still won't act as a page server. Better to leave it a transaction server, regardless of the system parameter. Later on, when transaction server will lose local storage, maybe there will be different server types - transaction and SA. To be determined.
```suggestion
  g_server_type = SERVER_TYPE_TRANSACTION;
```

File: msg/de_DE.utf8/cubrid.msg [hunk 2]
```diff
@@ -1338,7 +1338,7 @@ $ LOADDB
 1267 Cannot copy volume %s to %s. Error: %s.
 1268 At least one valid page server host is necessary, but no page server found in configuration.
 1269 At least one valid page server host is necessary, but no page server could be connected.
-1270 Error reserved for scalability development.
+1270 This tool (%1$s) cannot be executed on a server with remote storage enabled.
```
Comment (ribram) [inline msg/de_DE.utf8/cubrid.msg]:
```suggestion
1270 %1$s cannot be executed on a server with remote storage enabled.
```

Comment (cristiarg) [reply msg/de_DE.utf8/cubrid.msg]: Done

File: src/base/server_type.cpp (hunk 1 above)
Comment (cristiarg) [reply src/base/server_type.cpp]: Done

File: src/transaction/boot_sr.c [hunk 3]
```diff
@@ -5448,6 +5464,16 @@ xboot_emergency_patch (const char *db_name, bool recreate_log, DKNPAGES log_npag
       goto error_exit;
     }
 
+  {
+    const bool uses_remote_storage = prm_get_bool_value (PRM_ID_REMOTE_STORAGE);
+    if (uses_remote_storage)
+      {
+	er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_TOOL_INVALID_WITH_REMOTE_STORAGE, 1, "'emergency_patchlog'");
```
Comment (adrianbrab) [inline src/transaction/boot_sr.c:5471]: the indentation for this if block looks strange, is it git problem? If not I think it should be changed

**Review by adrianbrab — APPROVED** [review]

File: src/transaction/boot_sr.c (hunk 3 above)
Comment (cristiarg) [inline src/transaction/boot_sr.c:5471]: It's github's rendering of tabs at an arbitrary width.

**Pull request merged**